src/phpDocumentor/Descriptor/Builder/Reflector/Docblock/DescriptionAssemblerReducer.php

```diff
@@ -43,7 +43,7 @@ public function create(object $data, ?Descriptor $descriptor = null) : ?Descript
     /**
      * @param Tag[] $tags
      *
-     * @return TagDescriptor[]
+     * @return list<TagDescriptor|null>
```

Comment:
This is weird .. I would not expect nulls in this result. Perhaps we should resolve that :)

Reply:
I'll look into it

Reply:
Well, we go through buildDescriptor which has the filter feature inside so it can technically return null. If you say that it can't happen here, we can add an Assert to check that.

Reply:
Null can be used here, the new description implementation requires all array indexes to be there.
Since we do filter tags null can be a result.

Reply:
Hmm, would a Null Object not be a good fit here? I always get a bit uneasy with null values in arrays 😅

Reply:
Yes we could introduce that